src/load.ts

```diff
@@ -35,21 +36,36 @@ export const load = async (csvPath: string, column: string) => {
   // Extract the selected column from the CSV file
   const documents = data.map((row) => row[column] as string);
 
-  // Create a Pinecone index with the name "word-embeddings" and a dimension of 384
-  await createIndexIfNotExists(pineconeClient, indexName, 384);
+  // Get index name
+  const indexName = getEnv("PINECONE_INDEX");
+
+  // Check whether the index already exists. If it doesn't, create
+  // a Pinecone index with a dimension of 384 to hold the outputs
+  // of our embeddings model.
+  const indexList = await pinecone.listIndexes();
+  if (indexList.indexOf({ name: indexName }) === -1) {
+    await pinecone.createIndex({
+      name: indexName,
+      dimension: 384,
+      waitUntilReady: true,
```

Comment:
When `waitUntilReady: true` passed, the promise doesn't resolve until the index has status: ready. It handles the polling for status under the hood.